AzureTable/Microsoft.DataTransfer.AzureTable/ContinuationTokenParser.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Text;
+
+namespace Microsoft.DataTransfer.AzureTable
+{
+    /// <summary>
+    /// The class which is used to encode the continuation token
+    /// </summary>
+    public class ContinuationTokenParser
+    {
+        /// <summary>
+        /// Encode continuation token to fomart: (Version)!(TokenLength)!(CustomBase64EncodedToken)
+        /// </summary>
+        public static string EncodeContinuationToken(string key)
+        {
+            StringBuilder encodedContinuationToken = new StringBuilder();
+            // Version of the ContinuationToken
+            encodedContinuationToken.Append(1);
+            encodedContinuationToken.Append(exclamationDelimiter);
+
+            UTF8Encoding utf8Encoding = new UTF8Encoding();
+            string base64EncodedToken = Convert.ToBase64String(utf8Encoding.GetBytes(key.ToString()));
+            string customBase64EncodedString = UrlCustomEscapeBase64String(base64EncodedToken);
+
+            //Size is the lenght of base64 encoded key
+            encodedContinuationToken.Append(customBase64EncodedString.Length);
+            encodedContinuationToken.Append(exclamationDelimiter);
+
+            encodedContinuationToken.Append(customBase64EncodedString);
+            return encodedContinuationToken.ToString();
+        }
+
+        private static string UrlCustomEscapeBase64String(string token)
```

Comment:
I don't think the caller has a StringBuilder to pass in. And I don't like the way that calling a function might change the input parameter which might cause unnecessary confusion.

Reply:
Being a private method, this method is only called on line 23, where you have `encodedContinuationToken`. To eliminate the confusion you rename the method to `AppendXyz`, so that it clearly depicts that StringBuilder will be modified.

If for some other reason you don't want to optimize the code, then maybe you at least can pre-allocate the StringBuilder array, since the size of the `token` is known and all substitutions are character for character?

Reply:
Sorry, I thought you were talking about the public method. Yeah, that makes sense to me now.